rclpy/src/rclpy_common/src/common.c

```diff
@@ -109,6 +117,31 @@ rclpy_convert_to_py_names_and_types(rcl_names_and_types_t * names_and_types)
   return pynames_and_types;
 }
 
+static
+PyObject *
+_convert_rmw_time_to_py_duration(rmw_time_t * duration)
+{
+  uint64_t total_nanoseconds = RCUTILS_S_TO_NS(duration->sec) + duration->nsec;
+
+  PyObject * pyduration_module = PyImport_ImportModule("rclpy.duration");
+  if (!pyduration_module) {
+    return NULL;
+  }
+  PyObject * pyduration_class = PyObject_GetAttrString(pyduration_module, "Duration");
+  Py_DECREF(pyduration_module);
+  if (!pyduration_class) {
+    return NULL;
+  }
+
+  PyObject * args = PyTuple_New(0);
+  PyObject * kwargs = Py_BuildValue("{s:i}", "nanoseconds", total_nanoseconds);
```

Comment:
Need to check if `kwargs` is `NULL`. While `PyObject_Call()` allows kwargs to be `NULL`, that's not desired here. If `kwargs` is null it means `Py_BuildValue` set an exception, which would cause an [`assert()`  in `PyObject_Call`](https://github.com/python/cpython/blob/888f37bc2826d9ab2cbec6f153e7f58a34785c4a/Objects/call.c#L223).